src/lib/completion.ts

```diff
@@ -1,131 +1,144 @@
-import { Document } from 'langchain/document';
-
+//https://js.langchain.com/docs/modules/agents/how_to/custom_llm_agent
 import { ChatOpenAI } from 'langchain/chat_models/openai';
 import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
-import { SystemMessage, BaseMessage, AIMessage, HumanMessage } from 'langchain/schema';
-
-import { ChatHistoryType, type ChatHistory } from '$lib/history';
-import { BytesOutputParser } from 'langchain/schema/output_parser';
-
-import { QdrantClient } from '@qdrant/js-client-rest';
+import {
+	type AgentAction,
+	type AgentFinish,
+	HumanMessage,
+	type InputValues,
+	type AgentStep,
+	AIMessageChunk
+} from 'langchain/schema';
+
+import { type ChatHistory } from '$lib/history';
+import { formatLogToString } from 'langchain/agents/format_scratchpad/log';
+import { RunnableSequence } from 'langchain/schema/runnable';
+import { AgentExecutor } from 'langchain/agents';
+import { PREFIX, SUFFIX, TOOL_INSTRUCTIONS_TEMPLATE } from './prompt';
+import { getTools } from './tools';
+import { colors } from './colors';
 
 const DEFAULT_MODEL = 'gpt-3.5-turbo';
-const DEFAULT_COLLECTION = 'default';
 
 export class ChatbotCompletion {
-	private model: ChatOpenAI;
 	private embeddings_model: OpenAIEmbeddings;
-
-	private qdrant_client: QdrantClient;
-	private qdrant_collection: string;
+	private executor: AgentExecutor | undefined;
+	private openai_api_key: string;
+	private model_name;
+	private verbose: boolean;
 
 	constructor(
 		openai_api_key: string,
+		verbose: boolean,
 		{
-			openai_model = DEFAULT_MODEL,
-			qdrant_collection = DEFAULT_COLLECTION
```

Comment:
This is another reason to keep the qdrant collection in this class because I want someone developing locally to be able to use whatever collection they want as this allows multiple iterations on the database.